File: source/SEP3 - test (Rider)/WebServices Unit Test/Model/Game.cs
namespace Model
{
    public class Game
    {
        public int? Id { get; set; }
        public string Name { get; set; }
        public string Details { get; set; }
        public double Price { get; set; }
        public string User { get; set; }
        
        public string Type { get; set; }
        
        public string Genre { get; set; }
        
        public double? Rating { get; set; }
        
        public int? UserRating { get; set; }

        public override string ToString()
        {
            string temp = "Id: " + Id + "\nName: " + Name + "\nPrice: " + Price + "\nUser: " + User + "\nType: " + Type + "\nDetails: ";
            if (Details == null)
                temp += "null";
            else
                temp += Details;
            temp += "\nGenre: ";
            if (Genre == null)
                temp += "null";
            else
                temp += Genre;
            temp += "\nRating: ";
            if (Rating == null)
                temp += "null";
            else
                temp += Rating;
            temp += "\nUserRating: ";
            if (UserRating == null)
                temp += "null";
            else
                temp += UserRating;
            return temp;
        }
    }
}

File: source/SEP3 - tier 3 (IntelliJ)/src/main/java/model/Game.java
package model;

/**this class serves as a Data Object for Game*/
public class Game
{

    /**id of the game (nullable)*/
    public Integer Id;
    /**name of the game*/
    public String Name;
    /**price per day of the game*/
    public double Price;
    /**details of the game*/
    public String Details;
    /**owner of the game*/
    public String User;
    /**type of the game*/
    public String Type;
    /**genre of the game*/
    public String Genre;
    /**average rating of the game (nullable)*/
    public Double Rating;
    /**rating made by the user [depends on pov] (nullable)*/
    public Integer UserRating;

    /**initialises all instance variables*/
    public Game(Integer id, String name, double price, String details, String user, String type, String genre, Double rating, Integer userRating)
    {
        Id = id;
        Name = name;
        Price = price;
        Details = details;
        User = user;
        Type = type;
        Genre = genre;
        Rating = rating;
        UserRating = userRating;
    }

    public String toString()
    {
        return Id + " " + Name + " " + Price + " " + Details + " " + User + " " + Type + " " + Genre + " " + Rating + " " + UserRating;
    }
}